lib/path-utils.js

```diff
@@ -0,0 +1,169 @@
+import path from "path"
+import { promises } from "fs"
+const { stat, readFile } = promises
+
+import { unique } from "./utils"
+
+/** Converts given path to Posix (replacing \\ with /)
+ * @param {string} givenPath   Path to convert
+ * @returns {string}          Converted filepath
+ */
+export function posixifyPath(givenPath) {
+  return path.normalize(givenPath).replace(/\\/g, "/")
+}
+
+/** Check if the given path is a directory
+ * @param {string} givenPath absolute path
+ * @returns {Promise<boolean>}
+ */
+async function isDirectory(filepath) {
+  try {
+    return (await stat(filepath)).isDirectory()
+  } catch (error) {
+    return false
+  }
+}
+
+/**
+ * Checks if a provided path is a directory and returns a glob string matching
+ * all files under that directory if so, the path itself otherwise.
+ *
+ * Reason for this is that `glob` needs `/**` to collect all the files under a
+ * directory where as our previous implementation without `glob` simply walked
+ * a directory that is passed. So this is to maintain backwards compatibility.
+ *
+ * Also makes sure all path separators are POSIX style for `glob` compatibility.
+ *
+ * @param {string} givenPath The given path to be globified
+ * @param {string}   [process.cwd()]  The cwd to use to resolve relative pathnames
+ * @returns {Promise<string>} The glob path or the file path itself
+ */
+export async function globifyPath(givenPath, cwd = process.cwd()) {
+  const absolutePath = path.resolve(cwd, givenPath)
+  return posixifyPath(
+    (await isDirectory(absolutePath))
+      ? /* Globified directory */ globifyDirectory(absolutePath)
+      : /* Given path as is */ givenPath
+  )
+}
+
+/**
+ * Globifies a directory
+ * @param {string} givenDirectory The given directory to be globified
+ */
+export function globifyDirectory(givenDirectory) {
+  return posixifyPath(`${givenDirectory.replace(/[/\\]$/, "")}/**`)
+}
+
+/**
+ * Parse and globy the `.gitingore` file that exists in a directry
+ * @param {string} givenDirectory The given directory that has the `.gitignore` file
+ * @returns {Promise<Array<string>>} an array of glob patterns
+ */
+export async function globifyGitIgnoreFile(givenDirectory) {
+  return globifyGitIgnore(await readFile(path.join(givenDirectory, ".gitignore"), "utf-8"))
+}
+
+/**
+ * Read `.gitingore` file from a directry
+ * @param {string} gitIgnoreContent the content of the gitignore file
+ * @returns {Array<string>} an array of glob patterns
+ */
+export function globifyGitIgnore(gitIgnoreContent) {
```

Comment:
This has become a library of itself. I will make this an npm package later

Reply:
Ya that looks like a good candidate for a separate npm package.

Reply:
I am on the tail end of a project for work so it might be a month or so before I get around to writing test for this package.